internal/provider/waypoint/data_source_waypoint_application_template.go

```diff
@@ -123,6 +125,27 @@ func (d *DataSourceApplicationTemplate) Schema(ctx context.Context, req datasour
 					},
 				},
 			},
+			"variable_options": schema.ListNestedAttribute{
```

Comment:
I think it could be potentially useful in the future to have `user_editable` accessible from the data source, if you agree it is probably worth adding in here.